# Anderson and Galambos 2019

##MAGs
*anvio_bin_summary: The output of `anvi-summarize` for each sample.
*anvi_refine: The results of manual refinement of the MAGs in FS854 using `anvi-refine`
*anvio_MAG_meancov_data: The mean coverage of each MAG, as calculated by anvi'o
*bin_collections: Files describing which MAGs belong to which samples
*MCR_bin_fasta: FASTA files of the contigs contained in each MAG
*MCS_data: Module completion score (MCS) analysis for each MAG
##mapping
A folder for each sample that includes BAM files; individual profiles for each mapping created using anvi'o; and a merged profile for the whole sample.
##sample_data
*contigs: Contigs FASTA files for each sample and \*db files from anvio.
*DNA_reads
*RNA_reads
##scripts

This README is organized based on the figures and tables in the paper. Each figure/table gets its own section detailing which data were used, which scripts were used to create the figure, and run instructions + dependencies for those scripts. Please contact [Rika Anderson](mailto:randerson@carleton.edu) with any questions or bug fixes.


## Gene abundance across samples
### Data
### Scripts used
### How to run
### Dependencies
## Gene expression across samples
### Data
### Scripts used
### How to run
### Dependencies
## MAG expression across samples
### Data
### Scripts used
### How to run
### Dependencies
## Module completion score (MCS) across all MAGS
### Data
### Scripts used
### How to run
### Dependencies
## Tree of methanogenesis/ANME  genes
### Data
### Scripts used
### How to run
### Dependencies
## Table of average MCR across MAGS active at different samples
### Data
### Scripts used
### How to run
### Dependencies
>>>>>>> 8ced7a01bfe2e958a301c14d91118acf09623db8
